packages/field-base/test/shadow-focus-mixin.test.js

```diff
@@ -221,21 +212,49 @@ describe('shadow-focus-mixin', () => {
       customElement.appendChild(span);
 
       const spy = sinon.spy(focusElement, 'focus');
-      // Click center of span child element
-      const rect = span.getBoundingClientRect();
-      const middleX = Math.floor(rect.x + rect.width / 2);
-      const middleY = Math.floor(rect.y + rect.height / 2);
-      await sendMouse({ type: 'click', position: [middleX, middleY] });
-      // Clicking on some text content should not move focus
+      await click(span);
+      // Clicking on some text content should not move focus to focus element
       expect(spy.called).to.be.false;
     });
 
     it('should not delegate focus if the focus is not received from outside', () => {
-      const child = document.createElement('div');
-      customElement.appendChild(child);
+      customElement.focusElement.focus();
+      const spy = sinon.spy(focusElement, 'focus');
+      customElement.$.secondInput.focus();
+      customElement.$.thirdInput.focus();
+      customElement.$.secondInput.focus();
+      expect(spy.called).to.be.false;
+    });
 
-      focusin(customElement, child);
+    it('should skip host element in tab order when shift-tabbing', async () => {
+      const siblingInput = document.createElement('input');
+      customElement.parentElement.insertBefore(siblingInput, customElement);
 
+      customElement.focus();
+      expect(document.activeElement).to.equal(customElement);
```

Comment:
I'm not sure if it is the correct assertion. When calling `focus()` on the custom element, I expect that focus will be delegated to `#input` which is `focusElement`. However, it asserts that focus stays on the custom element.

I would rather test that it is possible to reach `siblingInput` from `#input` with <kbd>Shift</kbd> + <kbd>Tab</kbd>.

Reply:
Well it's kind of an opaque assumption. `customElement.focus()` should focus `customElement.focusElement` within the shadow DOM, as verified by another test. However `document.activeElement` will stay on `customElement`, because the focused element is in its shadow DOM.

But I've updated the test to use `customElement.focusElement.focus()` to make it more explicit.

Reply:
Oh, didn't notice that it was `document.activeElement`. Anyway, the new variant looks better to me. 👍